airflow/models/taskinstance.py

```diff
@@ -166,13 +170,23 @@ def clear_task_instances(
                 ti.max_tries = max(ti.max_tries, ti.prev_attempted_tries)
             ti.state = State.NONE
             session.merge(ti)
+
+        tr_filter.append((ti.dag_id, ti.task_id, ti.execution_date, ti.try_number))
```

Comment:
Should we use `TaskInstanceKey` and:
https://github.com/apache/airflow/blob/2a3960f359b1fcd5eed64fff472c4e463da74beb/airflow/models/taskinstance.py#L1987

To avoid code duplication?

Reply:
Hi @turbaszek. I looked into `filter_for_tis()`. It specifically creates a filter for `TaskInstance`. In this case, we need to filter for `TaskReschedule` instead. Also, it does not include a condition for `try_number`, which is necessary to get `TaskReschedule` right. 

```python
        return or_(
            and_(
                TaskInstance.dag_id == ti.dag_id,
                TaskInstance.task_id == ti.task_id,
                TaskInstance.execution_date == ti.execution_date,
            )
            for ti in tis
        )
```